pkg/dbmate/db.go

```diff
@@ -202,6 +218,23 @@ func doTransaction(db *sql.DB, txFunc func(Transaction) error) error {
 	return tx.Commit()
 }
 
+func transactAndRollback(db *sql.DB, txFunc func(Transaction) error) error {
+	tx, err := db.Begin()
+	if err != nil {
+		return err
+	}
+
+	if err := txFunc(tx); err != nil {
```

Comment:
YMMV, but it seems to be useful to preserve the transaction error message even if the rollback errors out, both are non-nil...when I have coded this same section in the past, I have done something like

```
return fmt.Errorf("Rollback err %s on transaction err %s",err1.Error(),err.Error())
```

Reply:
Love this suggestion!

Reply:
This is a good idea. Will do.

Reply:
done